test/gtest/common/mem_buffer.cc

```diff
@@ -45,17 +45,19 @@ void *mem_buffer::allocate(size_t size, ucs_memory_type_t mem_type)
 
     switch (mem_type) {
     case UCS_MEMORY_TYPE_HOST:
-        ptr = malloc(size);
+        ptr = calloc(1, size);
         if (ptr == NULL) {
             UCS_TEST_ABORT("malloc() failed");
         }
         return ptr;
 #if HAVE_CUDA
     case UCS_MEMORY_TYPE_CUDA:
         CUDA_CALL(cudaMalloc(&ptr, size));
+        CUDA_CALL(cudaMemset(ptr, 0, size));
```

Comment:
which test does not initialize the allocated buffer before sending it?
seems like test issue rather than allocator issue

Reply:
yes, this is the test issue
ok to split this PR into two ones?
- CMA fix
- valgrind error fix

for valgrind fix, i'd add `mem_buffer::memset` that fills a buffer according to the memory type (device/host memory). wdyt?

Reply:
created #4171